client/js/mousehandling.js

```diff
@@ -38,7 +38,15 @@ async function inputHandler(name, e) {
 
   if(target && target.id) {
     batchStart();
-    if(name == 'mousedown' || name == 'touchstart') {
+    if(!edit && (!jeEnabled || !e.ctrlKey) && widgets.get(target.id).passthroughMouse) {
+      if(name == 'mousedown' || name == 'touchstart') {
+        await widgets.get(target.id).mouseRaw('down', (coords[0] - roomRectangle.left)/scale, (coords[1] - roomRectangle.top)/scale);
```

Comment:
These three lines only differ by the event type now, right? Shouldn't be duplicated then.

I know, that's totally my fault. 😄

Reply:
Yes... and you had it in 4 places, spread out across about 40 lines of code.

Reply:
> These three lines only differ by the event type now, right? Shouldn't be duplicated then.
> 
> I know, that's totally my fault. 😄

Not sure if we should do it here, but might be best to refactor the code so that we have a variable that contains "down" if the event is "mousedown" or "touchstart", "move" if the event is "mousemove" or "touchmove", or "up" if the event is "mouseup" or "touchend". Not sure why lines 34-37 are needed for mouse events but not touch events.